src/ports/transak/component.ts

```diff
@@ -51,7 +51,8 @@ export function createTransakComponent(
       const cachedAccessToken = await cache.get<string>(TRANSAK_ACCESS_TOKEN_CACHE_KEY)
       if (!cachedAccessToken) {
         const { accessToken, expiresAt } = await getAccessToken()
-        await cache.set<string>(TRANSAK_ACCESS_TOKEN_CACHE_KEY, accessToken, fromMillisecondsToSeconds(expiresAt))
+        const keyTTL = expiresAt - Date.now()
+        await cache.set<string>(TRANSAK_ACCESS_TOKEN_CACHE_KEY, accessToken, fromMillisecondsToSeconds(keyTTL))
```

Comment:
Shall we update the `TRANSAK_ACCESS_TOKEN_CACHE_KEY` to prevent collisions with prior TTL?

Reply:
I haven't realized that the method the handler was executing was not forcing the re-fetch of the access token. I've modified it so it refreshes it forcefully, overriding the values of the previous key, thus setting the expiration key to the new one.